docs/tutorials/sessions/index.md

```diff
@@ -5,6 +5,87 @@ title: Sessions
 
 The following tutorials provide a detailed, step-by-step guide on how to integrate Account Abstraction Sessions. If you're developing in TypeScript, it's recommended to utilize the SDK.
 
+Before proceeding it is worth understanding an important data model that integrators are expected to understand prior to working with sessions (both single and batch). It is a low-level building block upon which sessions are built: the [Rule](#rules)
+
+```typescript
+/** Defines permissions for the args of an allowed method. */
+export interface Rule {
+  /**
+   * The offset in the ABI SVM contract helps locate the relevant data within the function call data
+   */
+  offset: number;
+  /**
+   * condition
+   *
+   * 0 - Equal
+   * 1 - Less than or equal
+   * 2 - Less than
+   * 3 - Greater than or equal
+   * 4 - Greater than
+   * 5 - Not equal
+   */
+  condition: number;
+  /** The value to compare against. */
+  referenceValue:
+    | string
+    | number
+    | bigint
+    | boolean
+    | ByteArray
+    | HardcodedReference;
+}
+
+/** Use the HardcodedReference type to bypass the auto-formatting which may mis-align your offset. */
+export type HardcodedReference = {
+  raw: Hex;
+};
+```
+
+## Rules
+
+Rules define permissions for the args of an allowed method.
+With rules, you can precisely define what should be the args of the transaction that is allowed for a given Session.
+Every Rule works with a single static arg or a 32-byte chunk of the dynamic arg.
+
+Since the ABI Encoding translates every static param into a 32-bytes word, even the shorter ones (like `address` or `uint8`), every Rule defines a desired relation (`Condition`) between n-th 32bytes word of the `calldata` and a reference Value (that is obviously a 32-bytes word as well).
```

Comment:
```suggestion
Since the ABI encoding translates every static parameter into a 32-byte word, even the shorter ones (like `address` or `uint8`), every Rule defines a desired relation (`Condition`) between n-th 32-byte word of the `calldata` and a reference Value (also a 32-byte word).
```